In [153]:
#import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns

In [154]:
data="C:/Users/Admin/Downloads/Order_Data_meriskill.csv" #import data from my local file path
mkt_data=pd.read_csv(data)
mkt_data.head()  #read the first five rows of the dataset


C:\Users\Admin\AppData\Local\Temp\ipykernel_13684\1317838774.py:2: DtypeWarning:

Columns (3,10) have mixed types. Specify dtype option on import or set low_memory=False.



OrderID          Region         Country CustID    Customer_Name  \
0  SO45080       Northwest   United States  14657      JOHN THOMAS   
1  SO45079       Southwest   United States  29255  KYLE WASHINGTON   
2  SO45082       Australia       Australia  11455        ROSS SANZ   
3  SO45081          Canada          Canada  26782       SETH LEWIS   
4  SO45083  United Kingdom  United Kingdom  14947   ALEJANDRO CHEN   

   ProductSKU Product_Category  OrderLineItem  OrderQuantity  ProductCost  \
0  BK-R50B-58           Plants              1            1.0     413.1463   
1  BK-R93R-48           Plants              1            1.0    2171.2942   
2  BK-M82B-44           Plants              1            1.0    1898.0944   
3  BK-R50B-44           Plants              1            1.0     413.1463   
4  BK-R93R-48           Plants              1            1.0    2171.2942   

  ProductPrice   OrderDate AcquisitionSource      TransactionID  Fraud  \
0     699.0982  01-01-2020        Meta-ads\r  I4127227587819694  False   
1      3578.27  01-01-2020      Google-ads\r  E9525730529761877  False   
2      3374.99  01-01-2020      Google-ads\r  Z6376643485803420  False   
3     699.0982  01-01-2020      Google-ads\r  R8609290025411138  False   
4      3578.27  02-01-2020        Meta-ads\r  Y9572700698484288  False   

  PaymentMethod  CardType Gender  
0    CREDITCARD  DISCOVER      M  
1    CREDITCARD        MC      M  
2    CREDITCARD      VISA      M  
3    CREDITCARD      VISA      M  
4    CREDITCARD      VISA      M

In [155]:
mkt_data.info()  #checks the type of datatypes in the dataset


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55910 entries, 0 to 55909
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   OrderID            55910 non-null  object 
 1   Region             55910 non-null  object 
 2   Country            55910 non-null  object 
 3   CustID             55910 non-null  object 
 4   Customer_Name      55907 non-null  object 
 5   ProductSKU         55910 non-null  object 
 6   Product_Category   55910 non-null  object 
 7   OrderLineItem      55910 non-null  int64  
 8   OrderQuantity      55910 non-null  float64
 9   ProductCost        55910 non-null  float64
 10  ProductPrice       55910 non-null  object 
 11  OrderDate          55910 non-null  object 
 12  AcquisitionSource  55909 non-null  object 
 13  TransactionID      55910 non-null  object 
 14  Fraud              55207 non-null  object 
 15  PaymentMethod      55910 non-null  object 
 16  CardType           506

In [156]:
mkt_data.shape   #checks the total number of rows and columns in the data

(55910, 18)

In [157]:
mkt_data.size #checks the size of the data

1006380

In [158]:
mkt_data.describe()  #performs descriptive statistics in the dataframe

OrderLineItem  OrderQuantity   ProductCost
count   55910.000000   55910.000000  55910.000000
mean        1.903237       1.501607    256.143102
std         1.021286       0.612305    496.563101
min         1.000000       1.000000      0.856500
25%         1.000000       1.000000      2.973300
50%         2.000000       1.000000     11.216300
75%         2.000000       2.000000     59.466000
max         8.000000       3.000000   2171.294200

Insights
1. The maximum productCost is 2171.294
2. The maximum number of OrderQuantity is 3 with an average of 1.5


In [159]:
mkt_data.duplicated()   #checks if the data has any duplicates

0        False
1        False
2        False
3        False
4        False
         ...  
55905    False
55906    False
55907    False
55908    False
55909    False
Length: 55910, dtype: bool

In [160]:
mkt_data.isnull().sum()   #checks for missing data or Nan values in the dataset

OrderID                 0
Region                  0
Country                 0
CustID                  0
Customer_Name           3
ProductSKU              0
Product_Category        0
OrderLineItem           0
OrderQuantity           0
ProductCost             0
ProductPrice            0
OrderDate               0
AcquisitionSource       1
TransactionID           0
Fraud                 703
PaymentMethod           0
CardType             5223
Gender               6328
dtype: int64

5 columns have  missing data. So, we will replace our missing data with zero

In [162]:
mkt_data=mkt_data.fillna(0)  #replaces missing data with zero
mkt_data

OrderID          Region         Country CustID    Customer_Name  \
0      SO45080       Northwest   United States  14657      JOHN THOMAS   
1      SO45079       Southwest   United States  29255  KYLE WASHINGTON   
2      SO45082       Australia       Australia  11455        ROSS SANZ   
3      SO45081          Canada          Canada  26782       SETH LEWIS   
4      SO45083  United Kingdom  United Kingdom  14947   ALEJANDRO CHEN   
...        ...             ...             ...    ...              ...   
55905  SO74143  United Kingdom  United Kingdom  28517    TROY GONZALEZ   
55906  SO74143  United Kingdom  United Kingdom  28517    TROY GONZALEZ   
55907  SO74143  United Kingdom  United Kingdom  28517    TROY GONZALEZ   
55908  SO74124          France          France  21676      VALERIE GUO   
55909  SO74124          France          France  21676      VALERIE GUO   

       ProductSKU    Product_Category  OrderLineItem  OrderQuantity  \
0      BK-R50B-58              Plants              1            1.0   
1      BK-R93R-48              Plants              1            1.0   
2      BK-M82B-44              Plants              1            1.0   
3      BK-R50B-44              Plants              1            1.0   
4      BK-R93R-48              Plants              1            1.0   
...           ...                 ...            ...            ...   
55905     WB-H098  Plant Care & Seeds              3            2.0   
55906     BC-R205  Plant Care & Seeds              2            1.0   
55907  BK-R19B-52              Plants              1            1.0   
55908     PK-7098  Plant Care & Seeds              2            2.0   
55909     TI-R092  Plant Care & Seeds              1            2.0   

       ProductCost ProductPrice   OrderDate AcquisitionSource  \
0         413.1463     699.0982  01-01-2020        Meta-ads\r   
1        2171.2942      3578.27  01-01-2020      Google-ads\r   
2        1898.0944      3374.99  01-01-2020      Google-ads\r   
3         413.1463     699.0982  01-01-2020      Google-ads\r   
4        2171.2942      3578.27  02-01-2020        Meta-ads\r   
...            ...          ...         ...               ...   
55905       1.8663         4.99  30-06-2022      Google-ads\r   
55906       3.3623         8.99  30-06-2022      Google-ads\r   
55907     343.6496       539.99  30-06-2022     Yt-Campaign\r   
55908       0.8565         2.29  30-06-2022      Google-ads\r   
55909       8.0373        21.49  30-06-2022      Google-ads\r   

           TransactionID  Fraud PaymentMethod  CardType Gender  
0      I4127227587819694  False    CREDITCARD  DISCOVER      M  
1      E9525730529761877  False    CREDITCARD        MC      M  
2      Z6376643485803420  False    CREDITCARD      VISA      M  
3      R8609290025411138  False    CREDITCARD      VISA      M  
4      Y9572700698484288  False    CREDITCARD      VISA      M  
...                  ...    ...           ...       ...    ...  
55905  Q7863980991839908  False    CREDITCARD        MC      M  
55906  T1770174949094173  False    CREDITCARD      VISA      M  
55907  Z1498721782917223  False    CREDITCARD      AMEX      M  
55908  J9171882375174307  False    CREDITCARD        MC      M  
55909  H1731861796382481  False        PAYPAL         0      0  

[55910 rows x 18 columns]

In [163]:
mkt_data.isnull().sum()  #check again if we still have missing data

OrderID              0
Region               0
Country              0
CustID               0
Customer_Name        0
ProductSKU           0
Product_Category     0
OrderLineItem        0
OrderQuantity        0
ProductCost          0
ProductPrice         0
OrderDate            0
AcquisitionSource    0
TransactionID        0
Fraud                0
PaymentMethod        0
CardType             0
Gender               0
dtype: int64

In [ ]:
DATA IS READY FOR ANALYSIS

EXPLORATORY DATA ANALYSIS